core/src/main/java/org/kiwix/kiwixmobile/core/di/components/CoreComponent.kt

```diff
@@ -79,6 +79,11 @@ interface CoreComponent {
   fun connectivityManager(): ConnectivityManager
   fun context(): Context
   fun downloader(): Downloader
+  fun bookmarksActivityComponent(): BookmarksActivityComponent.Builder
```

Comment:
We don't actually need all these subcomponents, the dependencies most of these are trying to resolve are all provided by CoreComponent already, you can most likely just add the `inject` methods for the activities and add the modules to the annotation here.

The only concern is if the scoping of the dependencies of the specific modules needs to be different. At that point it makes sense to have a subcomponent, you could also have the dependencies unscoped which would most likely produce the same effect